packages/back-end/src/models/OrganizationModel.ts

```diff
@@ -434,3 +467,97 @@ export async function updateMember(
     }),
   });
 }
+
+/**
+ * Delete an organization and all associated data
+ */
+export async function deleteOrganizationData(orgId: string) {
+  logger.info("Deleting org %s", orgId);
+
+  const org = await getOrganizationById(orgId);
+
+  if (!org) throw new Error("Organization not found");
+
+  const userIds = org.members.map((m: Member) => m.id);
+
+  const usersToDelete = [];
+  for (const userId of userIds) {
+    const allOrgsWithUser = await findOrganizationsByUserId(userId);
+    if (allOrgsWithUser.length === 1) usersToDelete.push(userId);
+  }
+
+  // these collections are not tied to orgs
+  const collectionsIgnored = ["agendaJobs", "installations"];
+
+  const allCollections = await mongoose.connection.db
+    .listCollections()
+    .toArray();
+
+  const collections = allCollections
+    .map((c) => c.name)
+    .filter((c) => !collectionsIgnored.includes(c));
```

Comment:
I know this approach makes the script more future-proof as we add new collections, but it does make me nervous for 2 reasons:

1. Performance. Some collections like `queries` have millions of rows on Cloud.  Because of the approach here, we need to run an `$or` query with a bunch of fields that don't exist.  This will make it do a complete table scan.  The delete query would block all write operations to the collection while it's running, further increasing load, so you could easily get into a feedback loop that ends up crashing the Mongo instance.
2. False positives.  We might add a new collection in the future that we should add to `collectionsIgnored`, but the dev adding it would have no idea this even exists.

The alternative approach is to explicitly list all collections here, along with the specific orgId column that they use.  This obviously has the downside of requiring us to update this list when we add a new collection.  But the pros might outweigh the cons here.